drivers/storage/portworx/deployment.go

```diff
@@ -0,0 +1,670 @@
+			args = append(args, "-k", strings.Join(t.cluster.Spec.Kvdb.Endpoints, ","))
+		}
+	}
+
+	if t.cluster.Spec.Network != nil {
+		if t.cluster.Spec.Network.DataInterface != nil &&
+			*t.cluster.Spec.Network.DataInterface != "" {
+			args = append(args, "-d", *t.cluster.Spec.Network.DataInterface)
+		}
+		if t.cluster.Spec.Network.MgmtInterface != nil &&
+			*t.cluster.Spec.Network.MgmtInterface != "" {
+			args = append(args, "-m", *t.cluster.Spec.Network.MgmtInterface)
+		}
+	}
+
+	if t.cluster.Spec.Storage != nil {
+		if t.cluster.Spec.Storage.Devices != nil {
+			for _, dev := range *t.cluster.Spec.Storage.Devices {
+				args = append(args, "-s", dev)
+			}
+		} else {
+			if t.cluster.Spec.Storage.UseAllWithPartitions != nil &&
+				*t.cluster.Spec.Storage.UseAllWithPartitions {
+				args = append(args, "-A")
+			} else if t.cluster.Spec.Storage.UseAll != nil &&
+				*t.cluster.Spec.Storage.UseAll {
+				args = append(args, "-a")
+			}
+		}
+		if t.cluster.Spec.Storage.JournalDevice != nil &&
+			*t.cluster.Spec.Storage.JournalDevice != "" {
+			args = append(args, "-j", *t.cluster.Spec.Storage.JournalDevice)
+		}
+		if t.cluster.Spec.Storage.SystemMdDevice != nil &&
+			*t.cluster.Spec.Storage.SystemMdDevice != "" {
+			args = append(args, "-metadata", *t.cluster.Spec.Storage.SystemMdDevice)
+		}
+
+	} else if t.cluster.Spec.CloudStorage != nil {
+		if t.cluster.Spec.CloudStorage.DeviceSpecs != nil {
+			for _, dev := range *t.cluster.Spec.CloudStorage.DeviceSpecs {
+				args = append(args, "-s", dev)
+			}
+		}
+		if t.cluster.Spec.CloudStorage.JournalDeviceSpec != nil &&
+			*t.cluster.Spec.CloudStorage.JournalDeviceSpec != "" {
+			args = append(args, "-j", *t.cluster.Spec.CloudStorage.JournalDeviceSpec)
+		}
+		if t.cluster.Spec.CloudStorage.SystemMdDeviceSpec != nil &&
+			*t.cluster.Spec.CloudStorage.SystemMdDeviceSpec != "" {
+			args = append(args, "-metadata", *t.cluster.Spec.CloudStorage.SystemMdDeviceSpec)
+		}
+		if t.cluster.Spec.CloudStorage.MaxStorageNodes != nil {
+			args = append(args, "-max_drive_set_count",
+				strconv.Itoa(int(*t.cluster.Spec.CloudStorage.MaxStorageNodes)))
+		}
+		if t.cluster.Spec.CloudStorage.MaxStorageNodesPerZone != nil {
+			args = append(args, "-max_storage_nodes_per_zone",
+				strconv.Itoa(int(*t.cluster.Spec.CloudStorage.MaxStorageNodesPerZone)))
+		}
+	}
+
+	if t.cluster.Spec.SecretsProvider != nil &&
+		*t.cluster.Spec.SecretsProvider != "" {
+		args = append(args, "-secret_type", *t.cluster.Spec.SecretsProvider)
+	}
+
+	if t.startPort != defaultStartPort {
+		args = append(args, "-r", strconv.Itoa(int(t.startPort)))
+	}
+
+	if t.imagePullPolicy != v1.PullAlways {
+		args = append(args, "--pull", string(t.imagePullPolicy))
+	}
+
+	if t.cluster.Annotations[annotationCSIVersion] != "" {
+		args = append(args, "--csiversion", t.cluster.Annotations[annotationCSIVersion])
+	}
+
+	if t.isPxDev {
+		args = append(args, "--dev")
+	}
+
+	if t.isPKS {
+		args = append(args, "--keep-px-up")
```

Comment:
Can you checkout with @zoxpx. The latest oci-monitor might have --keep-px-up behavior as the default. So we may not needed this.

Reply:
As discussed added a comment. Will keep it for older version of oci-mon